deface/deface.py

```diff
@@ -58,6 +59,13 @@ def draw_det(
             frame[y1:y2, x1:x2] = resized_replaceimg
         elif replaceimg.shape[2] == 4:  # RGBA
             frame[y1:y2, x1:x2] = frame[y1:y2, x1:x2] * (1 - resized_replaceimg[:, :, 3:] / 255) + resized_replaceimg[:, :, :3] * (resized_replaceimg[:, :, 3:] / 255)
+    elif replacewith == 'mosaic':
+        for y in range(y1, y2, mosaicsize):
+            for x in range(x1, x2, mosaicsize):
+                pt1 = (x, y)
+                pt2 = (min(x2, x + mosaicsize - 1), min(y2, y + mosaicsize - 1))
+                color = (int(frame[y, x][0]), int(frame[y, x][1]), int(frame[y, x][2]))
+                cv2.rectangle(frame, pt1, pt2, color, -1)
```

Comment:
This works but on small `mosaicsize` values this can slow down the process due to the high number of nested loop iterations and draw `cv2` draw calls. Do you have an idea on how to optimize this? The loop nesting can be avoided with `itertools.product()` but the real problem is the number of resulting `cv2.rectangle()` calls. I haven't found a way to draw multiple rectangles at once with the `cv2` API yet.

Reply:
I'll still merge this PR as is because it's a nice addition, it works and is reasonably fast on your default value of 20. Feel free to open another PR if you get around to optimizing this code.